lang/cs/Org.Apache.REEF.Tang/Formats/ConfigurationFile.cs

```diff
@@ -133,12 +133,33 @@ public static HashSet<string> ToConfigurationStringList(IConfiguration c)
                 } 
                 else 
                 {
-                    Org.Apache.REEF.Utilities.Diagnostics.Exceptions.Throw(new IllegalStateException(), LOGGER);
+                    throw new BindException($"Failed to serialize set of unsupported type {e.Value.GetType()}");
                 }
                 
                 l.Add(GetFullName(e.Key) + '=' + Escape(val));
             }
 
+            foreach(var kvp in conf.GetBoundList())
+            {
+                foreach (var item in kvp.Value)
+                {
+                    string val = null;
+                    if (item is string)
+                    {
+                        val = GetFullName((string)item);
+                    }
+                    else if (kvp.Value is INode)
```

Comment:
shouldn't it be `item is INode` instead of `kvp.Value is INode`? also, can use `if (item is INode node) ...` syntax to avoid the cast